src/lib.rs

```diff
@@ -274,6 +274,7 @@ pub fn make_test(config: &Config, testpaths: &TestPaths) -> test::TestDescAndFn
             name: make_test_name(config, testpaths),
             ignore: early_props.ignore,
             should_panic: should_panic,
+            allow_fail: false,
```

Comment:
This should likely be a field of the `Config` structure so that it's configurable.